File: ios/Classes/Serializer/LegalLinksSettingsSerializer.swift
import UsercentricsUI

extension LegalLinksSettings {
    static func from(enumString: String?) -> LegalLinksSettings? {
        switch enumString {
        case "FIRST_LAYER_ONLY":
            return LegalLinksSettings.firstLayerOnly
        case "SECOND_LAYER_ONLY":
            return LegalLinksSettings.secondLayerOnly
        case "BOTH":
            return LegalLinksSettings.both
        case "NONE":
            return LegalLinksSettings.none
        default:
            return nil
        }
    }
}

File: lib/src/internal/serializer/legal_links_settings_serializer.dart
import 'package:usercentrics_sdk/src/model/legal_links_settings.dart';

class LegalLinksSettingsSerializer {
  static dynamic serialize(LegalLinksSettings? value) {
    switch (value) {
      case null:
        return null;
      case LegalLinksSettings.firstLayerOnly:
        return 'FIRST_LAYER_ONLY';
      case LegalLinksSettings.secondLayerOnly:
        return 'SECOND_LAYER_ONLY';
      case LegalLinksSettings.both:
        return 'BOTH';
      case LegalLinksSettings.hidden:
        return 'HIDDEN';
    }
  }
}
